cpp/inc/bond/ext/grpc/shared_unary_call.h

```diff
@@ -5,64 +5,52 @@
 
 #include <bond/core/config.h>
 
-#include <bond/ext/grpc/detail/unary_call_impl.h>
-#include <bond/ext/grpc/unary_call.h>
+#include "detail/unary_call_impl.h"
+#include "unary_call.h"
 
 #include <utility>
 
-namespace bond { namespace ext { namespace gRPC {
-
-/// @brief A shared owner of the details of a single async, unary call.
-///
-/// Call \ref Finish or \ref FinishWithError to send a response back to the
-/// client.
-///
-/// To obtain a shared_unary_call instance, call \ref unary_call::share or
-/// construct a shared_unary_call from a \ref unary_call rvalue (e.g., via
-/// moving).
-///
-/// If no explicit call to \p Finish or \p FinishWithError has been made before all
-/// the shared owners are destroyed, a generic internal server error is
-/// sent.
-template <typename Request, typename Response>
-class shared_unary_call final : public detail::unary_call_base<Request, Response>
+namespace bond { namespace ext { namespace gRPC
 {
-    using base_type = typename shared_unary_call::unary_call_base;
-
-public:
-    /// @brief Creates an empty shared_unary_call.
```

Comment:
`shared_unary_call` was intentionally designed to be default constructible so that is can be used in places where a default constructible object is wanted or needed. What do we need to do make it default constructible again?

Yes, something like `boost::optional<unary_call>` could be used here, but our users *explicitly asked* for default constructibility.

The motivating use case is as a data member of a struct that holds per-request state, but the data member needs to be created before the service dispatches to the business logic.

How about we make `unary_call` default constructible too?